monocle/gitlab/mergerequest.py

```diff
@@ -0,0 +1,532 @@
+                )
+            ],
+            repository_prefix="group",
+            repository_fullname="group/dummyrepo",
+            repository_shortname="dummyrepo",
+            author=create_ident(url, "John", []),
+            committer=create_ident(url, "John", []),
+            merged_by=create_ident(url, "Zuul", []),
+            branch="dummy-feature",
+            target_branch="main",
+            created_at="2020-04-11T07:00:15Z",
+            merged_at="2020-04-12T07:00:15Z",
+            updated_at="2020-04-12T07:00:15Z",
+            closed_at="2020-04-12T07:00:15Z",
+            state="MERGED",
+            duration=3600,
+            mergeable=None,
+            labels=None,
+            assignees=None,
+            approval=None,
+            draft=None,
+            self_merged=False,
+        )
+        dummy_event = Event(
+            _id="cce_1",
+            _type="ChangeCreatedEvent",
+            created_at="2020-04-11T07:00:15Z",
+            author=create_ident(url, "John", []),
+            repository_prefix="group",
+            repository_fullname="group/dummyrepo",
+            repository_shortname="dummyrepo",
+            url="https://dummy.com/change/id",
+            branch="dummy-feature",
+            target_branch="main",
+            number=1,
+            change_id="group/dummyrepo/1",
+            on_author=create_ident(url, "John", []),
+            on_created_at="2020-04-11T07:00:15Z",
+            changed_files=[SimpleFile(path="README.md")],
+            approval=None,
+        )
+        objects.extend([dummy_change, dummy_event])
+        return objects
+
+
+if __name__ == "__main__":
+    import argparse
+    import json
+    import os
+    import sys
+    from pprint import pprint
+
+    from monocle.gitlab import graphql
+
+    parser = argparse.ArgumentParser(prog="mergerequest")
+
+    parser.add_argument("--crawler", help="run crawler", action="store_true")
+    parser.add_argument(
+        "--updated-since", help="stop date for the crawler (YYYY-mm-dd)"
+    )
+    parser.add_argument("--loglevel", help="logging level", default="INFO")
+    parser.add_argument(
+        "--url",
+        help="GitLab service url",
+        default="https://gitlab.com",
+    )
+    parser.add_argument(
+        "--group",
+        help="GitLab group",
+        required=True,
+    )
+    parser.add_argument("--repository", help="The repository or regexp", required=True)
+    parser.add_argument("--id", help="The merge request id")
+
+    args = parser.parse_args()
+
+    logging.basicConfig(
+        level=getattr(logging, args.loglevel.upper()),
+        format="%(asctime)s - %(name)s - %(threadName)s - "
+        + "%(levelname)s - %(message)s",
+    )
+
+    tg = TokenGetter(args.url, args.group, args.token)
+
+    mrf = MRsFetcher(
+        graphql.GitlabGraphQLQuery(token_getter=tg),
+        args.url,
+        args.group,
+        idents_config=[],
+        repository=args.repository,
```

Comment:
why are some with `key=value` and some don't?

asking out of ignorance (and because it looks a little odd) and would like to get an answer

Reply:
This was a good comment. Fixed that.